### DISCUSSION TITLE
Display and update JSON via TextArea form component?

### DISCUSSION THREAD

**Discussion by nickspringham** [filamentphp/filament#2284] [Help] [answered]
Hi 👋 

I have a multidimensional JSON string in the db, which is cast to JSON in the model.

I'm utilising the TextArea form component to allow the admin to view and update the JSON.
But the data is coming in as: [object Object]
<img width="813" alt="filament" src="https://user-images.githubusercontent.com/45168010/166667316-65855a37-76e0-49cd-bfca-06ea26b3e831.png">

Any advice on how to get the json_decoded string into the component?

**Comment by danharrin** [comment, marked as answer]
The issue here is the cast. Try this:

```php
->afterStateHydrated(fn (TextArea $component, $state) => $component->state(json_decode($state)))
```

**Reply by nickspringham** [reply]
I knew it would be some sort of casting issue.
I have added it and got the following TypeError:
`json_decode(): Argument #1 ($json) must be of type string, stdClass given`

**Reply by danharrin** [reply]
Make sure that the cast is `array` and not `json` in the model. I know. It sounds backwards.

`array` casts to a PHP array, which is what you'd expect
`json` casts to an StdClass 🤨

**Reply by ryangjchandler** [reply]
I think you actually want `json_encode`, not `json_decode` since you want to show the string version.

**Reply by danharrin** [reply]
ah that could be it

**Reply by nickspringham** [reply]
Appreciate your help guys! worked a treat 👍

**Comment by jBernavaPrah** [comment]
Can this example be included in the documentation? It took me three whole hours to find it! :)
